scm/driver/bitbucket/linker.go

```diff
@@ -24,8 +24,19 @@ func (l *linker) Resource(ctx context.Context, repo string, ref scm.Reference) (
 	case scm.IsPullRequest(ref.Path):
 		d := scm.ExtractPullRequest(ref.Path)
 		return fmt.Sprintf("%s%s/pull-requests/%d", l.base, repo, d), nil
-	case ref.Sha == "":
+	case scm.IsBranch(ref.Path) && ref.Sha == "":
 		t := scm.TrimRef(ref.Path)
+
+		// Bitbucket has a bug where the "source view" link for
+		// a branch which contains a slash results in a 404.
+		// The link to the "branch view" works with names containing
+		// a slash so we do this for branches with slashes in its names.
+		// See https://jira.atlassian.com/browse/BCLOUD-14422 for more information
+		//
+		if scm.BranchContainsSlash(t) {
```

Comment:
since this function is only used by Bitbucket, it should be private an internal to the Bitbucket package, as opposed to a library-level function.

Reply:
also, this does not really need to be a separate function, since it is just wrapping the container function with no extra logic.

```diff
-if scm.BranchContainsSlash(t) {
+if strings.Contains(t, "/") {
```